app/javascript/src/components/Player.tsx

```diff
@@ -202,6 +209,29 @@ class Player extends React.Component<PlayerProps, PlayerState> {
     )
   }
 
+  private goBack() {
+    return (
+      <a className="SlantButton" id="back-button" onClick={this.setPreviousState.bind(this)} >
+        Back
+      </a>
+    )
+  }
+
+  private setPreviousState() {
+    let previous = this.state.previousState
+    previous.splice(-1)
```

Comment:
i have a few questions here:
1. what does `splice` do again?
2. I think it's bad form to modify a direct reference to state, which is why there are `clone`s all over this class, did you find that that wasn't necessary?
3. you store `this.state.previousState` in a variable here (`previous`) but then later in `setState` you ignore the variable and reference `this.state.previousState` again. Is there any way to make use of one variable all the way down so future devs don't have to hold all the state logic in their head reading this function?

Reply:
1. splice (-1) just removes the last element. 
2. again just forgot the way it was done the first time. can change that to `clone`
3. i'll go ahead and fix that.